plugins/modules/ec2_vpc_nat_gateway.py

```diff
@@ -614,6 +624,9 @@ def create(client, module, subnet_id, allocation_id, tags, client_token=None,
         changed = True
         return changed, result, msg
 
+    if connectivity_type == "private":
+        params.pop('AllocationId')
```

Comment:
The fact we set the `AllocationId` key and we delete it 20 lines below is a bit inelegant.